crates/extension_host/src/wasm_host.rs

```diff
@@ -376,6 +403,10 @@ impl WasmHost {
                 },
             );
 
+            // Store will yield after 1 tick, and get a new deadline of 1 tick after each yield.
+            store.set_epoch_deadline(1);
+            store.epoch_deadline_async_yield_and_update(1);
```

Comment:
Here there are a few other options: 
1. Change either the epoch interval or deadline to just put an upper bound on component execution (this is what Spin does)
2. Do the callback version where you can decide if you want to yield or trap

No matter which way, since I believe you all have a timeout further up, the future will eventually get dropped, but any variation should still help.